cscs-checks/apps/amber/amber_check.py

```diff
@@ -3,95 +3,190 @@
+
+    @run_after('setup')
+    def set_perf_patterns(self):
         self.perf_patterns = {
-            'perf': sn.extractsingle(r'ns/day =\s+(?P<perf>\S+)',
-                                     output_file, 'perf', float, item=1)
+            self.benchmark: sn.extractsingle(r'ns/day =\s+(?P<perf>\S+)',
+                                             self.output_file, 'perf',
+                                             float, item=1)
         }
-        self.maintainers = ['SO', 'VH']
-        self.tags = {'scs', 'external-resources'}
-
-
-@rfm.parameterized_test(*(
-    [variant, arch, scale]
-    for variant in ['prod', 'maint']
-    for arch in ['CPU', 'GPU']
-    for scale in ['small', 'large']
-    if (not (scale, arch) == ('large', 'GPU') and
-        not (variant, arch) == ('maint', 'CPU'))
-))
-class AmberCheck(AmberBaseCheck):
-    def __init__(self, variant, arch, scale):
-        super().__init__('mdin.%s' % arch, 'amber.out')
-        self.descr = 'Amber parallel %s %s check (%s)' % (scale, arch, variant)
-        self.tags |= {'maintenance' if variant == 'maint' else 'production'}
-        if arch == 'GPU':
-            self.valid_systems = ['daint:gpu', 'dom:gpu']
-            self.executable = 'pmemd.cuda.MPI'
-            self.reference = {
-                'dom:gpu': {
-                    'perf': (30.0, -0.05, None, 'ns/day')
-                },
-                'daint:gpu': {
-                    'perf': (30.0, -0.05, None, 'ns/day')
-                },
-            }
-            if variant == 'prod' and scale == 'small':
-                self.tags.add('health')
-        elif arch == 'CPU':
-            self.valid_systems = ['daint:mc']
-            if scale == 'small':
-                self.valid_systems += ['dom:mc']
-
-            self.executable = 'pmemd.MPI'
-            self.strict_check = False
-            if scale == 'small':
-                self.num_tasks = 216
-                self.num_tasks_per_node = 36
-                self.reference = {
-                    'dom:mc': {
-                        'perf': (8.0, -0.05, None, 'ns/day')
-                    },
-                    'daint:mc': {
-                        'perf': (7.6, -0.05, None, 'ns/day')
-                    }
-                }
+
+
+@rfm.simple_test
+class AmberGPUCheck(AmberBaseCheck):
+    num_tasks = 1
+    num_tasks_per_node = 1
+    num_gpus_per_node = 1
+    valid_systems = ['daint:gpu', 'dom:gpu']
+    executable = 'pmemd.cuda.MPI'
+    input_file = 'mdin.GPU'
+    output_file = 'amber.out'
+    descr = f'Amber GPU check'
+    tags = {'maintenance', 'production', 'health'}
+    reference = REFERENCE_GPU_PERFORMANCE
+
+
+@rfm.simple_test
+class AmberCPUCheck(AmberBaseCheck):
+    scale = parameter(['small', 'large'])
+    valid_systems = ['daint:mc', 'eiger:mc']
+    executable = 'pmemd.MPI'
+    input_file = 'mdin.CPU'
+    output_file = 'amber.out'
+    tags = {'maintenance', 'production'}
+    reference = REFERENCE_CPU_PERFORMANCE_LARGE
```

Comment:
It feels a bit weird that the `REFERENCE_CPU_PERFORMANCE_LARGE` is the default one. Since you have the hook `set_perf_reference` below already, I'd set the `reference` in this hook for all cases.
```suggestion
```